### PR TITLE
Code suggestions

### PR DESCRIPTION
**Pull request by muffato**
Here I'm proposing two changes:

- The code for computing the duration works in a toy example - but I haven't tested it in TreeVal
- I think `github` is not a great variable name. What about simply on using `config_profile_name` ?

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/sanger-tol/treeval/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/hic_mapping.nf
Comment (DLBPointon) [inline subworkflows/local/hic_mapping.nf:1]: I agree, that is much better.

File: lib/TreeValProject.groovy
Comment (DLBPointon) [inline lib/TreeValProject.groovy:1]: I tried something similar to this when working on it, however, the dreaded compilation error rears it's head.
Running this on our test dataset, I get:

```
ERROR ~ BUG! exception in phase 'semantic analysis' in source unit 'Script_24208ff5' The lookup for TreeValProject caused a failed compilation. There should not have been any compilation from this call.
```

It seems as though any arithmetic going on in this file causes a compilation error.

Comment (DLBPointon) [reply lib/TreeValProject.groovy:1]: Actually even just replacing `new Date().format("yyyy-MM-dd HH:mm:ss.SSSZ")` with `OffsetDateTime.now()` causes a failure.

Comment (muffato) [reply lib/TreeValProject.groovy:1]: Nevermind ! Let's just leave it as it was.

For the record, here was my toy example
```nextflow
#!/usr/bin/env nextflow
nextflow.enable.dsl=2 

import java.time.OffsetDateTime

process sayHello {
  input: 
    val x
  output:
    stdout
  script:
    def d = OffsetDateTime.now()
    def duration = Duration.between( workflow.start, d )
    """
    echo '$x world! $duration'
    """
}

workflow {
  Channel.of('Bonjour', 'Ciao', 'Hello', 'Hola') | sayHello | view
}
```

Comment (DLBPointon) [reply lib/TreeValProject.groovy:1]: I got it!

Most of your code did work, I am apparently just blind to imports so using.....

```bash
import java.time.OffsetDateTime;
import java.time.Duration;

class TreeValProject {
    //
    // Generates a small summary containing context for the input files
    // Creates a new file containing this context + pipeline_execution data
    // Will be used for graph generation.
    //

    public static void summary(workflow, params, metrics, log) {

        def date_completed = OffsetDateTime.now()

        def input_data = [:]
        input_data['version']           = NfcoreTemplate.version( workflow )
        input_data['runName']           = workflow.runName
        input_data['session_id']        = workflow.sessionId
        input_data['duration']          = Duration.between( workflow.start, date_completed ).toSeconds()
        input_data['DateStarted']       = workflow.start
        input_data['DateCompleted']     = date_completed
        input_data['entry']             = params.entry

```

produces: 
```markdown
Pipeline_duration:  16
Pipeline_datastrt:  2023-09-15T16:31:26.057655375+01:00
Pipeline_datecomp:  2023-09-15T16:31:42.581488030+01:00
```

**Review by DLBPointon — CHANGES_REQUESTED** [review]
It works!

Just a couple of changes. Thanks for the help @muffato

File: lib/TreeValProject.groovy
Comment (muffato) [reply lib/TreeValProject.groovy:1]: Yeah ! 💪🏼

**Pull request merged**